packages/salesforcedx-apex-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -596,12 +593,14 @@ export class ApexReplayDebug extends LoggingDebugSession {
     sourceLine?: number,
     category = 'stdout'
   ): void {
-    if (msg && msg.length !== 0) {
+    if (msg?.length !== 0) {
       const event: DebugProtocol.OutputEvent = new OutputEvent(
         `${msg}${EOL}`,
         category
       );
-      event.body.source = sourceFile;
+      if (sourceFile) {
+        event.body.source = sourceFile;
+      }
       event.body.line = sourceLine;
```

Comment:
1. Why did you add `is (sourceFile) {` around `event.body.source = sourceFile`?  What is the harm in setting the `sourceFile` variable when the `printToDebugConsole()` function is called without the `sourceFile` argument and `sourceFile` automatically gets the value of `undefined`?
2. If there is indeed an edge case covered here, why didn't you add `is (sourceLine) {` around `event.body.line = sourceLine`?  Why do `sourceFile` and `sourceLine` need to be treated differently?

Reply:
After discussion in slack I moved the validation to userDebugState.ts:35